### DISCUSSION TITLE
Create Mount entity

### DISCUSSION THREAD

**Discussion by Distractic** [Minestom/Minestom#1598] [Help]
I would like to create Mount creature where the Player can control it (Move, jump etc.)

I checked the #829 but the code is not working now, the different fields cannot be found.

Someone has an idea on how to implement a creature mount ?

Thanks in advance

**Comment by Am-Gone** [comment]
To mount the Player on an Entity you can use [Entity#addPassenger(Player)](https://javadoc.minestom.net/net/minestom/server/entity/Entity.html#addPassenger(net.minestom.server.entity.Entity))

Now in order to control the entity you'll have to get the [players' vehicle information ](https://javadoc.minestom.net/net/minestom/server/entity/Player.html#getVehicleInformation()) and modify the velocity of the entity based on that every ticks. You can either override the tick method of the player or make a task with the scheduler.

The code should be something like this:
```
if(player.getVehicleInformation().getForward() > 0) { // Player is going forward
    mount.setVelocity(player.getPosition().direction().withY(0).mul(2));
} else if(player.getVehicleInformation().getForward() < 0) { // Player is going backwards
    mount.setVelocity(player.getPosition().direction().withY(0).mul(-2));
}
```

Of course you'll have to tweak the values of the vectors, although I don't know if this code will work as I'm not able to test it right now
